plugins/reporters/cyclonedx/src/main/kotlin/CycloneDxReporter.kt

```diff
@@ -252,6 +257,30 @@ class CycloneDxReporter : Reporter {
         return outputFiles
     }
 
+    private fun addVulnerabilitiesToBom(input: ReporterInput, bom: Bom) {
```

Comment:
I think an argument like `advisorVulnerabilities:  Map<Identifier, List<Vulnerability>>` holds the data that's needed, since I also need the identifier to populate the `affects` field with [references](https://cyclonedx.org/docs/1.5/json/#vulnerabilities_items_affects_items_ref) pointing to the `bom-ref` components fields.

Reply:
Right! In essence, I was meaning to say to make the function take the return value of `input.ortResult.getVulnerabilities()` instead of the whole `input`, to make more clear that the implementation really only cares about vulnerability data.